gomock/call.go

```diff
@@ -75,6 +75,9 @@ func (c *Call) Do(f interface{}) *Call {
 	return c
 }
 
+// Return sets the values to return when the target method is called.
+//
+// If Return and Do are called, the values returned via Do will be returned.
```

Comment:
Correct me if I'm wrong, but this PR seems to make it so that if Do is called period, regardless of whether Return is also called, the return values of the Do func will be returned, making this comment technically inaccurate.

This also breaks if Do has a different return signature than the parent called function.  The lambda provided to Do must already have the same input argument signature as the parent function, else the reflection call panics, but the current design doesn't require that of the return value of the Do function, and since that return value isn't currently used, I'd be willing to be most Dofuncs in the wild don't return anything.

However, with this design, if the lambda provided to Do returns a different number of or types of return values, the generated code will either panic due to a slice out of bounds on the ret (if the Dofunc returns fewer return values than the parent function), or will silently fail and return a zero value, since the generated code uses a two-value type assertion, but discards the boolean.  This makes the changes here problematic, because all existing usages of Do will trigger a panic if they have a Dofunc that returns nothing and a parent function that has more than zero return values.

On a behavioral side, an explicit call to Return should probably override the return-what-Do-returned behavior (if you're explicitly calling Return, it's because you want the function to return something specific), and the code in general should probably be updated to only return the Dofunc return values if they match the number and type of the parent function itself.